style.md

```diff
@@ -0,0 +1,672 @@
+# Jsdoom Style Guide
+
+The purpose of this style guide is to keep jsdoom's source code readable, approachable, consistent, and maintainable. When contributing to jsdoom, please follow these guidelines. Pull requests containing code that does not follow the guidelines may be rejected until the code has been made conformant.
+
+These are not strict rules! Exceptions are allowed where it improves readability. However, exceptions should be just that: Exceptional. Exceptions should be made sparingly, if they are made at all.
+
+Please [create a GitHub issue](https://github.com/pineapplemachine/jsdoom/issues/new) if you would like to request a change or an addition to this style guide.
+
+Throughout this guide, specific examples of conforming code will be provided like so:
+
+``` ts
+// An example of conforming code
+console.log("jsdoom is pretty neat");
+```
+
+
+
+# Formatting
+
+## Indent using four spaces
+
+Code in jsdoom is indented with four spaces. The line followed by an open brace `{` should be indented one level further than the line with the open brace. The line with the corresponding closing brace `}` should be indented one level less.
+
+``` ts
+// [Comment explaining the purpose of my function]
+function myFunction(): void {
+    doStuff();
+    if(condition){
```

Comment:
if/while/for statements should be padded with spaces around the parentheses and braces IMHO.

Reply:
Recently I've been considering adopting spaces between the close paren and the open brace. For a long time I've written code without that space, but I am starting to see the logic. Though I still think a space between the keyword and the open paren looks strange, and seems inconsistent with function invocations which almost everyone writes without a space before the open paren, i.e. `myFunction(myArgs)`.

Really, I deliberately left out any specific guidelines regarding whitespace in these situations because this is one of the few areas where I actually _don't_ have a strong opinion...

I would be open to enforcing either of these styles:

``` ts
if(condition){
    doStuff();
}
```

``` ts
if(condition) {
    doStuff();
}
```

I don't feel as comfortable enforcing this style:

``` ts
if (condition) {
    doStuff();
}
```

Reply:
For now, any of the above styles are allowed. The style guide may be amended later to prefer one style over the others.